src/sparse/impl/KokkosSparse_spgemm_impl_kkmem.hpp

```diff
@@ -1352,6 +1355,11 @@ void
 	  chunksize = min_hash_size; //this is for used hash indices
 	  chunksize += min_hash_size ; //this is for the hash begins
 	  chunksize += max_nnz; //this is for hash nexts
+    // adjust for aligned accesses
+    while ((chunksize-2*min_hash_size) % sizeof(nnz_lno_t) != 0)
```

Comment:
I couldn't convince myself. Can chunksize be odd here due to `max_nnz`? If so and `nnz_lno_t` is even, this would be an infinite loop since chunk size would always be odd.

Reply:
@jjwilke I rushed this in, need to add safety though I don't think this could happen based on the types that are used to select these values.